internal/clearingway/menu.go

```diff
@@ -0,0 +1,198 @@
+package clearingway
+
+import (
+	"fmt"
+
+	"github.com/Veraticus/clearingway/internal/discord"
+	"github.com/bwmarrin/discordgo"
+)
+
+// types of UI elements
+type MenuType string
+const(
+	MenuMain      MenuType = "menuMain"
+	MenuVerify    MenuType = "menuVerify"
+	MenuRemove    MenuType = "menuRemove"
+	MenuEncounter MenuType = "menuEncounter"
+)
+
+type CommandType string
+const(
+	CommandMenu CommandType = "menu"
+	CommandSub1 CommandType = "sub1"
+	CommandSub2 CommandType = "sub2"
+	CommandSub3 CommandType = "sub3"
+)
+
+// struct to hold data for all different menu components
+type Menus struct {
+	Menus    map[string]*Menu
+}
+
+type Menu struct {
+	Name           string  // internal name to uniquely identify menus
+	Type           MenuType  // type of menu to differentiate AdditionalData types
+	Title          string  // title to show in embed
+	Description    string  // optional description to show in embed
+	ImageURL       string  // optional image URL
+	AdditionalData MenuAdditionalData  // additional data depending on MenuType
+}
+
+type MenuAdditionalData interface{}
+
+type MenuRoleHelper struct {
+	Role         *Role
+	Prerequisite *Role
+}
+
+type MenuMainData struct {
+	Message *discordgo.MessageSend
+}
+
+type MenuStaticData struct {
+	Message *discordgo.InteractionResponse
+}
+
+type MenuEncounterData struct {
+	Roles        map[string]*MenuRoleHelper
+	ExtraRoles   []*Role
+	RoleType     []RoleType
+	MultiSelect  bool
+	RequireClear bool
+}
```

Comment:
I'll do this but keep in mind the different types of menus will have completely unrelated fields as additional data if we do make one mega struct. I'll have to keep `MenuRoleHelper` separate though since that's a different struct